src/constants/permissions.js

```diff
@@ -0,0 +1,5 @@
+export default {
+  PUBLIC: { role: 'public', value: 0 },
```

Comment:
What's the point of `value`?
I think the following structure is more straightforward.
{
 roles: {
    ADMIN: { value: 'admin', priority: 1 },
    MODERATOR: { value: 'moderator', priority: 2 },
    USER: { value: 'user', priority: 3 }
 }
}